red_october/models/red_october_user.py

```diff
@@ -0,0 +1,206 @@
+            return res
+        for vault in res.vault_ids:
+            with vault.get_api() as api:
+                api.create_user()
+                if res.is_admin:
+                    record._update_role('admin')
+        res.active = True
+        return res
+
+    @api.multi
+    def update(self, vals):
+        """ It updates the user on the remote vaults. """
+        for record in self:
+            role = None
+            active = vals.get('is_active')
+            is_admin = vals.get('is_admin')
+            if is_admin is not None and is_admin != record.is_admin:
+                role = 'admin'
+            elif active is not None and active != record.active:
+                role = 'revoke'
+            if role:
+                record._update_role(role)
+        return super(RedOctoberUser, self).update(vals)
+
+    @api.multi
+    def unlink(self):
+        """ It removes the user from the remote vaults. """
+        self._update_role('revoke')
+        return super(RedOctoberUser, self).unlink()
+
+    @api.model
+    def change_current_user(self, ro_user_id):
+        """ It changes the current session user to the provided.
+
+        Args:
+            ro_user_id (int): ID of the RedOctoberUser to add as the current
+                session default.
+        """
+        setattr(request.session, self.SESSION_USER_ATTR, ro_user_id)
+
+    @api.multi
+    def change_password(self, old_passwd, new_passwd, confirm_passwd):
+        """ It changes the password for the user singleton. """
+        self.ensure_one()
+        if new_passwd != confirm_passwd:
+            raise ValidationError(_(
+                'The new password and confirmation do not match. Please '
+                'try again.',
+            ))
+        user = self.get_current_user()
+        for vault in self.vault_ids:
+            with vault.get_api(user, old_passwd) as api:
+                api.change_password(new_passwd)
+        return True
+
+    @api.model
+    def get_current_user(self):
+        """ It returns the RedOctoberUser that the session user is using.
+
+        This method currently returns the default selected in the user, but
+        plans are to allow for the control of this via session.
+
+        Returns:
+            RecOctoberUser: The user that is currently in use for this
+                session.
+        """
+        user = None
+        try:
+            user = self.browse(
+                getattr(request.session, self.SESSION_USER_ATTR),
+            )
+        except AttributeError:
+            pass
+        if not user:
+            user = self.upsert_by_user()
+        return user
+
+    @api.model
+    def read_current_user(self):
+        user = self.get_current_user()
+        return user and user.read() or {}
+
+    @api.model
+    def get_user_profiles(self):
+        """ It returns the current user's profiles. """
+        users = self.search([
+            ('user_id', '=', self.env.user.id),
+        ])
```

Comment:
Interesting - how could I use `self.mapped` here? `self.filtered` perhaps?

General question on this - I assumed these methods first gathered all records, then filtered/mapped. Are they operating directly within the DB, before iterating the cursor?